### PR TITLE
Support config and cache file

### PR DESCRIPTION
**Pull request by pbrisbin**
If using codeclimate-wrapper, manage config.yml and cache.yml files in the
user's home directory (following the XDG spec). These files are currently used
to manage the recently introduced version check feature.

Just re-opening #587 against master.

### CODE REVIEW

**Comment by pbrisbin** [comment]
Will merge and release tomorrow. Confirmed locally this doesn't break anything, but there is a JSON parse error visible in DEBUG, so it's just skipping the check at the moment.

/cc @pointlessone

**Comment by pointlessone** [comment]
JSON? Is Verssions API broken?

**Comment by pbrisbin** [comment]
Must've been temporary, it's working now.

Only thing left here now is the test failures, I guess.

**Comment by pbrisbin** [comment]
I don't see a persistent UUID written anywhere though,

```
% cat ~/.{config,cache}/codeclimate/*.yml
---
latest-version: 0.52.0
outdated: false
last-version-check: 2017-03-15 14:03:52.801191019 +00:00
```

Is that expected?

**Comment by pointlessone** [comment]
No, the config doesn't appear to be saved.

**Comment by pbrisbin** [comment]
@pointlessone it appears to me nothing ever calls `GlobalCache#save`. The other file is saves as part of all the write methods.

File: lib/cc/cli/version_checker.rb [hunk 1]
```diff
@@ -7,19 +10,33 @@ class VersionChecker
       DEFAULT_VERSIONS_URL = "https://versions.codeclimate.com".freeze
 
       def check
+        return unless global_config.check_version?
```
Comment (pointlessone) [inline lib/cc/cli/version_checker.rb]: We need to add `global_config.save` after this line.

Comment (pointlessone) [inline lib/cc/cli/version_checker.rb]: Sorry, after the next one.

**Comment by pointlessone** [comment]
@pbrisbin Fixed specs and added config persistence. Separate commit for ease of review. Safe to squash before merging.

**Comment by pbrisbin** [comment]
Thanks! Will try to do a release today.

**Pull request merged**